script.js

```diff
@@ -1,4 +1,300 @@
+var playerTwoChoose = function () {
+  player2Number = 0;
+
+  // Computer auto-generates number for user depending on input
+  if (userInputs[inputIndex] == "highest") {
+    // Sort the dice rolls's values in decreasing order
+    player2DiceRolls.sort(function (a, b) {
+      return b - a;
+    });
+    var counter = 0;
+    while (counter < numberOfDices[numberOfDiceIndex]) {
+      player2Number = player2Number + String(player2DiceRolls[counter]);
+      counter += 1;
+    }
+  }
+
+  if (userInputs[inputIndex] == "lowest") {
+    // Sort the dice rolls's values in ascending order
+    player2DiceRolls.sort(function (a, b) {
+      return a - b;
+    });
+    var counter = 0;
+    while (counter < numberOfDices[numberOfDiceIndex]) {
+      player2Number = player2Number + String(player2DiceRolls[counter]);
+      counter += 1;
+    }
+  }
+
+  // Add number result to array
+  player2NumberList.push(Number(player2Number));
+
+  // Generate score
+  player2Score = generateScore(player2NumberList);
+
+  // Generate output text
+  var outputWord = `Player 2, the computer has auto-generated the
+    ${userInputs[inputIndex]} combined number for you, which is ${player2NumberList[numberIndex]} based on the dice rolls ${player2DiceRolls}. <br><br>`;
+
+  if (userInputs[inputIndex] == "highest") {
+    var winnerText = decideWinner();
+  } else if (userInputs[inputIndex] == "lowest") {
+    var winnerText = decideWinnerByLowestNumber();
+  }
+
+  // Include leaderboard that lists the 2 players and their scores in decreasing order
+  if (player1Score > player2Score) {
+    var outputText = `LEADERBOARD <br><br> Player 1 - Score: ${player1Score} <br> Player 2 - Score: ${player2Score}`;
+  } else {
+    var outputText = `LEADERBOARD <br><br> Player 2 - Score: ${player2Score} <br> Player 1 - Score: ${player1Score}`;
+  }
+
+  outputText =
+    outputWord +
+    `You chose '${userInputs[inputIndex]} combined number mode'. <br><br> The winner of this round: ${winnerText}.<br><br><br>` +
+    outputText +
+    `<br><br><br>OTHER STATISTICS <br><br> Player 1 - Current Number: ${player1NumberList[numberIndex]} <br> Player 1 - All Numbers: ${player1NumberList} <br><br> Player 2 - Current Number: ${player2NumberList[numberIndex]} <br> Player 2 - All Numbers: ${player2NumberList} <br><br><br> Play again by inputting either "highest" to play the 'highest combined number mode' or "lowest" to play the 'lowest combined number mode'`;
+
+  // Restart the game or play again
+  gameMode = GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST;
+
+  // Update dice index count
+  diceIndex += 2;
+  inputIndex += 1;
+  numberIndex += 1;
+  numberOfDiceIndex += 1;
+
+  // Redefine values of number to 0
+  player1Number = 0;
+  player2Number = 0;
+
+  return outputText;
+};
+
+// Decide who the winner is (i.e. whose number is higher)
+var decideWinner = function () {
+  if (player1Number > player2Number) {
+    return "Player 1";
+  } else {
+    return "Player 2";
```

Comment:
since you are returning in the if statement already, you can skip the else statement and just write return player 2 after the if statement. This is because for scenarios where the if condition is fulfilled JS will exit after it returns Player 1 and not run line 238.

An illustration of how the if statement in the function could be is 

function ()={
     if (player1Number>player2Number){
          return 'Player 1'
     }
     return 'Player 2'
 }